backend/src/main/java/com/crm/service/CarServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.crm.service;
+
+import com.crm.dto.mapper.CarMapper;
+import com.crm.dto.response.CarResponse;
+import com.crm.model.db.CarEntity;
+import com.crm.repository.CarRepository;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class CarServiceImpl implements CarService {
+
+    private final CarRepository carRepository;
+    private final CarMapper carMapper;
+
```

Comment:
What about @Autowired?

Reply:
Autowired is not required - Spring automatically Inject dependencies in the constructor (provided that there is only one in a class). [https://stackoverflow.com/questions/41092751/spring-injects-dependencies-in-constructor-without-autowired-annotation](url)

But yeah, maybe it is better to add @Autowired above the constructor for the future purposes if we add a non-parameter constructor too.

Reply:
Thank you for explanation ;)